### PR TITLE
Spa

### PR DESCRIPTION
**Pull request by neverwannafly**
Create a basic app structure

### CODE REVIEW

File: interview-scheduler/src/app.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+let router = new Router();
+
+const renderPage = async () => {
+  let view = router.match();
+  console.log(view);
+  const root = document.getElementById('root');
+  if (view != null) {
+    root.innerHTML = await view.render();
+    await view.postRender();
```
Comment (ngudbhav) [inline interview-scheduler/src/app.js:14]: By name, post render is a function which is supposed to be called after every render. Can we do something to call postRender like a callback or something?

**Comment by ngudbhav** [comment]
Very Good implementation of SPA! Keep up.

File: interview-scheduler/src/services/Utils.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        request.id          = r[2]
+        request.verb        = r[3]
+
+        return request
+    }
+
+    // --------------------------------
+    //  Simple sleep implementation
+    // --------------------------------
+    , sleep: (ms) => {
```
Comment (ngudbhav) [inline interview-scheduler/src/services/Utils.js:24]: Why do we need this?

File: interview-scheduler/src/views/components/Modals.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            </div>
+          </div>
+        </div>
+      </div>
+    `;
+    return view;
+  },
+  postRender: async () => {
+    const $findUsers = document.getElementById('find-users');
+    $($findUsers).selectize({
```
Comment (ngudbhav) [inline interview-scheduler/src/views/components/Modals.js:115]: Vanilla JS in 1 line and jQuery in 2nd. :-p

File: interview-scheduler/src/views/pages/Error404.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+          </center>
+        </div>
+      </div>
+    `;
+    return view;
+  },
+  postRender: async () => {
+    const navbar = document.getElementById('navbar');
+    navbar.innerHTML = await Navbar.render();
+    await Navbar.postRender();
```
Comment (ngudbhav) [inline interview-scheduler/src/views/pages/Error404.js:20]: Should navbar be rendered before this page or after this page?

File: interview-scheduler/src/views/pages/Index.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import Navbar from "../components/Navbar.js";
+import Modals from "../components/Modals.js";
+
+
+let Index = {
+  name: "Index",
+  render : async () => {
```
Comment (ngudbhav) [inline interview-scheduler/src/views/pages/Index.js:7]: async?

File: interview-scheduler/src/services/Utils.js (hunk 2 above)
Comment (neverwannafly) [reply interview-scheduler/src/services/Utils.js:24]: Removed it.

File: interview-scheduler/src/views/components/Modals.js (hunk 3 above)
Comment (neverwannafly) [reply interview-scheduler/src/views/components/Modals.js:115]: Sorry for the inconsistencies :P 
I'll try to fix it!

File: interview-scheduler/src/views/pages/Error404.js (hunk 4 above)
Comment (neverwannafly) [reply interview-scheduler/src/views/pages/Error404.js:20]: the order in which the elements are rendered doesn't matter as I'm defining a root for every element in the HTML itself. So they will always get rendered at the place they are supposed to no matter where I render them.

**Pull request merged**